File: backend/src/main/java/com/ll/nbe342team8/domain/qna/question/dto/PostResQuestionDto.java
package com.ll.nbe342team8.domain.qna.question.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.ll.nbe342team8.domain.member.member.entity.Member;
import com.ll.nbe342team8.domain.qna.question.entity.Question;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public record PostResQuestionDto(List<QuestionDto> questions) {
    public PostResQuestionDto(Member member) {
        this(List.copyOf(member.getQuestions().stream()
                .map(QuestionDto::new)
                .collect(Collectors.toList()))); // 🔥 불변 리스트 적용
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/qna/question/dto/PostResQuestionDto.kt
package com.ll.nbe342team8.domain.qna.question.dto

import com.ll.nbe342team8.domain.member.member.entity.Member


data class PostResQuestionDto( val questions: List<QuestionDto>) {
    constructor(member: Member) : this(
        member.questions.map { QuestionDto(it) }.toList()
    )
}